src/crystal/system/thread.cr

```diff
@@ -45,6 +45,9 @@ class Thread
   # Returns the Fiber representing the thread's main stack.
   getter! main_fiber : Fiber
 
+  # Returns the Fiber currently running on the thread.
+  property! current_fiber : Fiber
```

Comment:
Does the comment need to mention that it will be the last executing fiber if there are no currently executing fibers, ie the program is waiting on the event loop on io or the run_loop if it is waiting for more fibers?

Reply:
At least with the current scheduler, there is always an active, executing fiber. Even if it's waiting on something.
So I don't think we need any specific mention as of now.
This might change with the MT refactor (https://github.com/crystal-lang/rfcs/pull/2) at which point we'll need to revisit the documentation.

Reply:
Good point. We might also consider resetting it back to `nil` before `#start` terminates :thinking:

Reply:
Oh, but there must always be a current fiber while the thread is running, because the thread needs a running stack (even when preempted or waiting), and we still need to switch between fibers when we want to resume a fiber, at least until the thread has terminated (in which case we should reset it back to `nil`: no current fiber before it starts nor after it stops).

For RFC 2 I made it so that when we'd block, the scheduler resumes the thread's main fiber, which resumes the scheduler's run loop (i.e. the thread's main fiber becomes the current fiber).

Reply:
And now I realize that `Thread#start` is very likely GC unsafe.